### PR TITLE
Fix TagsInput duplicate deletion behavior

### PR DESCRIPTION
**Pull request by ctdunc**
fixes https://github.com/mantinedev/mantine/issues/6873

### CODE REVIEW

**Comment by rtivital** [comment]
Thanks!

File: packages/@mantine/core/src/components/TagsInput/TagsInput.tsx [hunk 1]
```diff
@@ -336,7 +336,7 @@ export const TagsInput = factory<TagsInputFactory>((_props, ref) => {
       key={`${item}-${index}`}
       withRemoveButton={!readOnly}
       onRemove={() => {
-        setValue(_value.filter((i) => item !== i));
+        setValue(_value.splice(index, 1));
```
Comment (adelamarAtGalileo) [inline packages/@mantine/core/src/components/TagsInput/TagsInput.tsx:339]: Is this correct? I'm seeing that clicking X to remove a tag now removes all other tags except the target

Comment (ctdunc) [reply packages/@mantine/core/src/components/TagsInput/TagsInput.tsx:339]: think I should've used `.toSpliced` here @adelamarAtGalileo can you let me know if this resolves it? 

Seems alright to me, but clearly I missed it yesterday. Storybook must not have refreshed :shrug:.

Comment (adelamarAtGalileo) [reply packages/@mantine/core/src/components/TagsInput/TagsInput.tsx:339]: `splice` modifies the array and returns the removed value

Screenshot 2024-09-30 at 1 55 04 PM

`toSpliced` doesn't modify the array and returns the remaining values

Screenshot 2024-09-30 at 1 57 00 PM

So yes, `.toSpliced` should work here

Comment (max-degterev) [reply packages/@mantine/core/src/components/TagsInput/TagsInput.tsx:339]: nice, hopefully released soon, already got complaints from users :D

Comment (adelamarAtGalileo) [reply packages/@mantine/core/src/components/TagsInput/TagsInput.tsx:339]: @max-degterev You can downgrade to  `"@mantine/core": "^7.11.1"` you can avoid the newly introduced bug while waiting for a new release with the fix

Comment (max-degterev) [reply packages/@mantine/core/src/components/TagsInput/TagsInput.tsx:339]: maybe @rtivital pushes this out soon, I keep my hopes up :)

**Pull request merged**